dts/bindings/stepper/gpio-stepper-controller.yaml

```diff
@@ -0,0 +1,40 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO Stepper Controller cluster for darlington transistor arrays or dual H-bridge
+  stepper {
+    compatible = "gpio-steppers";
+    status = "okay";
+    motor:motor {
+      gpios = <&gpioa 9 GPIO_ACTIVE_HIGH>,
+              <&gpioc 7 GPIO_ACTIVE_HIGH>,
+              <&gpiob 0 GPIO_ACTIVE_HIGH>,
+              <&gpioa 7 GPIO_ACTIVE_HIGH>;
+    };
+  };
+
+compatible: "gpio-steppers"
+
+child-binding:
+  description: GPIO Controller for stepper motor
+  include:
+    - name: stepper-controller.yaml
+      property-allowlist:
+        - micro-step-res
+
+  properties:
```

Comment:
would it make sense to add support for an optional `enable` pin? testing this code out on a driver that has such a pin, no need to do that in this PR, may be a followup

Reply:
follow up would be a step_dir (DRV8825)driver, where i want to test/introdzce this :)